### PR TITLE
Scatterpolargl fixes

### LINKED ISSUE #3094: flipping between scatterpolar and scatterpolargl causes traces to disappear
This Pen shows the behaviour: https://codepen.io/nicolaskruchten/pen/ReVyjv

### PR DESCRIPTION
**Pull request by etpinard**
fixes #3094 and brings `scatterpolargl` axis expansion on-par with `scatterpolar`.

cc @plotly/plotly_js

### CODE REVIEW

File: src/traces/scatterpolargl/index.js [hunk 1]
```diff
@@ -26,22 +27,28 @@ function calc(container, trace) {
...
-    calcColorscales(trace);
+    if(len < rArray.length) rArray = rArray.slice(0, len);
+    if(len < thetaArray.length) thetaArray = thetaArray.slice(0, len);
 
     stash.r = rArray;
     stash.theta = thetaArray;
 
-    trace._extremes.x = Axes.findExtremes(radialAxis, rArray, {tozero: true});
+    // We could add TOO_MANY_POINTS logic like in Scattergl.calc,
+    // if users asks for scaterpolargl charts with > 1e5 pts
```
Comment (alexcjohnson) [inline src/traces/scatterpolargl/index.js]: isn't that more or less the purpose of `scatterpolargl`? I'd vote to just add it now.

File: src/plots/polar/index.js [hunk 2]
```diff
@@ -66,6 +69,11 @@ function clean(newFullData, newFullLayout, oldFullData, oldFullLayout) {
                 oldSubplot.clipPaths[k].remove();
             }
         }
+
+        if(mustCleanScene && oldSubplot._scene) {
+            oldSubplot._scene.destroy();
+            oldSubplot._scene = null;
+        }
```
Comment (alexcjohnson) [inline src/plots/polar/index.js]: Does it work correctly if there's a separate `scattergl` or `scatter3d` subplot when you flip between `scatterpolar` and `scatterpolargl`?

Comment (etpinard) [reply src/plots/polar/index.js]: Yes it does work correctly :lock: in -> https://github.com/plotly/plotly.js/pull/3098/commits/cab319d48b4d602223127842f9ccea84273b62bc with some explanations.

File: src/traces/scatterpolargl/index.js (hunk 1 above)
Comment (etpinard) [reply src/traces/scatterpolargl/index.js]: done in https://github.com/plotly/plotly.js/pull/3098/commits/25fb7dc583b0d2e2b424d23add79a18636f35b9f

**Comment by etpinard** [comment]
Ready for review again.

File: src/traces/scattergl/index.js [hunk 3]
```diff
@@ -129,13 +126,10 @@ function calc(gd, trace) {
 
     // stash scene ref
     stash._scene = scene;
-    stash.index = scene.count;
+    stash.index = scene.count++;
```
Comment (alexcjohnson) [inline src/traces/scattergl/index.js]: this of course works... but I can't say I *like* using `++` mid-assignment

File: src/traces/scatterpolargl/index.js [hunk 4]
```diff
@@ -20,11 +20,11 @@ var subTypes = require('../scatter/subtypes');
 
 var TOO_MANY_POINTS = require('../scattergl/constants').TOO_MANY_POINTS;
 
-function calc(container, trace) {
-    var layout = container._fullLayout;
+function calc(gd, trace) {
+    var fullLayout = gd._fullLayout;
```
Comment (alexcjohnson) [inline src/traces/scatterpolargl/index.js]: ha yeah, that's certainly clearer! 🍰

**Comment by alexcjohnson** [comment]
Beautiful! With the slight exception of https://github.com/plotly/plotly.js/pull/3098#discussion_r227873310 but I won't hold up the 💃 for that.
Love the new trace type test.

**Pull request merged**